### ISSUE TITLE
[Bug]:  Registers an unload listener

### ISSUE THREAD

**Issue by GaurangMistry** [dnnsoftware/Dnn.Platform#5913]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What happened?

We have many sites in DNN and we are creating all sites perfectly with Google LightHouse. In Google LightHouse, we are facing one issue common in all sites is "Registers an unload listener". This is coming from Default Jquery file which DNN used.

I searched in google and got many suggestions to delete "Unload" from addEventListener method from Jquery or use "visibilitychange" but it's not work. We also take help of Jquery experts but not get expected result. So, we wish if you have any idea about it and you can directly work with us on project to fix this issue.

### Steps to reproduce?

1. Install Google Chrome Light House plugin
2. Open website: https://www.shopsmart.website/ in Google Chrome
3. Right-Click on website and select "Inspect". 
4. There is one option called "Lighthouse". Select it and click on "Analyze page load".
5. Select "Best Practices" and you will get error for "Registers an unload listener".

### Current Behavior

![2024-01-10_16-09-04](https://github.com/dnnsoftware/Dnn.Platform/assets/33444274/f8034585-c02e-4198-8d8a-9ae9eb1b200c)
![2024-01-10_16-08-40](https://github.com/dnnsoftware/Dnn.Platform/assets/33444274/dffcf62b-2362-491a-8309-9a7e593a1fc7)
![2024-01-10_16-07-58](https://github.com/dnnsoftware/Dnn.Platform/assets/33444274/b59f40d2-d7fb-4f82-ac9c-364b66a72992)

### Expected Behavior

Need to fix "Registers an unload listener" error and get 100% score in Google Lighthouse

### Relevant log output

```shell
I searched in google and got many suggestions to delete "Unload" from addEventListener method from Jquery or use "visibilitychange" but it's not work. We also take help of Jquery experts but not get expected result.
```

### Affected Versions

9.13.0 (latest release)

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by uzmannazari** [comment]
hello valadas i thinks this is not the issue for jquery or ajax. this is the problem that dnn used unload listeners on backend i think!
there is way to use beforeunload to fix this problem but i dont know how to change and where the functions used this

**Comment by uzmannazari** [comment]
but this is the big problem that dnn websites are now have low rate of lighthouse because of this problem

**Comment by valadas** [comment]
From what I can see, it looks like a Microsoft.ajas issue and you already opened an issue there too https://github.com/MicrosoftDocs/feedback/issues/3975

**Comment by valadas** [comment]
Searching the codebase here, I found this: https://github.com/dnnsoftware/Dnn.Platform/blob/463b867bcdd850aa9bb6a581f6d8b4da59b7f648/DNN%20Platform/Website/js/Debug/MicrosoftAjaxWebForms.js#L1181 which may or may not be related

**Comment by 6TELOIV** [comment]
It appears to be a MicrosoftAjax issue, but no idea where an appropriate place would be to open an issue with them.

**Comment by valadas** [comment]
Well, the link above is in DNN, I don't know off-the-cuff if it is the only instance, but it would be a start... I am not sure the error is on Microsoft side now with searching here...

**Comment by uzmannazari** [comment]
as i see this is to replace the unload with beforeunload function but i dont know what will happen if we do that on dnn backend and whats the usage of this function !

**Comment by 6TELOIV** [comment]
That could be something to try out and just see what happens; modify [this line](https://github.com/dnnsoftware/Dnn.Platform/blob/463b867bcdd850aa9bb6a581f6d8b4da59b7f648/DNN%20Platform/Website/js/Debug/MicrosoftAjax.js#L3592) to say `beforeunload` instead of `unload`, though that probably won't work if the event actually needs to fire because [`beforeunload` events won't run code](https://stackoverflow.com/questions/4802007/window-onbeforeunload-not-working-in-chrome).

I would recommend trying `pagehide`, or a more complex change would be `visibilitychange` because you'd have to check if the visibility is now "hidden" before firing their event. See this [Google Dev article](https://developer.chrome.com/docs/web-platform/deprecating-unload#alternatives_to_unload_events) for more info.

**Comment by 6TELOIV** [comment]
Another thing to try is just delete the line entirely and see what happens. It already doesn't fire reliably, and it will be gone for good eventually, so it would be good to see what happens if it just wasn't there at all.

**Comment by uzmannazari** [comment]
i changed the MicrosoftAjax.js file yesterday but the scriptresource.axd file didnt changed!
i dont know how to change this ScriptResource.axd file and how this file is being created!

**Comment by valadas** [comment]
scriptresource.axd is a handler that can serve multiple scripts potentially merged together and/or minified, it is not a file exactly. What needs to be done is to see what is using onload, figure out the effects of changing it, etc.

**Comment by valadas** [comment]
Also, I spotted by a search one usage, but there could be more to it...

**Comment by GaurangMistry** [comment]
Hi @valadas @uzmannazari 

Because of use of Microsoft Ajax this issues comes. After long investigation, I handled this in different way.
Actually, DNN use this Microsoft Ajax file to handle Admin panel after login as a SuperUser or Admin but DNN load this Microsoft Ajax every time on page. This is actual issue. 

To handle this, I set condition on Default.aspx page 
---------------------------------------
if(!HttpContext.Current.User.Identity.IsAuthenticated)
{
        AJAX.GetScriptManager(this).AjaxFrameworkMode = AjaxFrameworkMode.Disabled;
}
---------------------------------------

If no any user logged into site, then Disabled this Microsoft Ajax.

When we used DNN installer package, on that time we have no  Default.aspx.cs file to write this code. So, I get this page from DNN source code and set this condition. 
But it would be great if DNN handle this part in their coding. 

Thanks

**Comment by uzmannazari** [comment]
great @GaurangMistry 
thanks for share this. but how can we set this condition on default.aspx file?

**Comment by uzmannazari** [comment]
![Untitled](https://github.com/dnnsoftware/Dnn.Platform/assets/52697118/e3ab0ba7-c8d4-4777-9c33-ef8726acfbd8)
i have this error when set this condition on default.aspx file

**Comment by valadas** [comment]
I don't think this is the proper fix. not using the offending method would be a better fix. The code is compiled into an assembly. There could be some workaround but the proper fix is to find the correct source of the bug.

**Comment by uzmannazari** [comment]
I checked the @GaurangMistry website and now there is no error for unload listener on his website!

**Comment by uzmannazari** [comment]
@valadas , this error on google lighthouse is not bug! it is about the standards that to be better on google websites need that

**Comment by valadas** [comment]
I agree, my point is that preventing the ajax library from loading is more of a workaround than a proper fix. I may have ramifications and the proper fix would be to fix the underlying code to not use unload.

**Comment by GaurangMistry** [comment]
@uzmannazari Default.aspx.cs code is coming from DNN assembly which @valadas mentioned and that's same thing I mentioned in my previous comments.
So, I added Default2.aspx page and use that page codefile in Default.aspx. Please check below screenshot.

![2024-03-19_10-45-31](https://github.com/dnnsoftware/Dnn.Platform/assets/33444274/9af569eb-8ba0-44e0-bf3a-7adb0c9bc7db)

And I write that code into Default2.aspx.cs. I know this is just trick to fix this but If DNN add this code in Default.aspx.cs code then it will fix in direct DNN assembly and we no need to do this type of settings.

**Comment by uzmannazari** [comment]
your right mr valadas, now what should we do to fix this problem with lighthouse? 
should we edit microsoft ajax js or dnn core?

**Comment by valadas** [comment]
I am not sure without spending the time looking into it. I _think_ in DNN usage of microsoft Ajax potentially.

**Comment by uzmannazari** [comment]
hello valadas,
did you find any way to solve this problem? 
there is a link that have a way how to fix but i think the ScriptResource.axd is not related to microsoftajax.js because editing this file not apply!

https://gtmetrix.com/avoid-unload-event-listeners.html

**Comment by valadas** [comment]
I did not spend time on that no.

**Comment by uzmannazari** [comment]
do you know microsoftajax is need on user side or not?
I think mr @GaurangMistry has disabled it on useside but not on login page because of lighthouse report on his website.

do you have any idea about that?

**Comment by valadas** [comment]
I don't know off-the-cuff
